### ISSUE TITLE
ELB documentation does not match reality: listeners are mandatory

### ISSUE THREAD

**Issue by minac** [ansible/ansible-modules-core#4227]
##### ISSUE TYPE
- Documentation Report
##### COMPONENT NAME

ec2_elb_lb
##### ANSIBLE VERSION

ansible 2.1.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
##### CONFIGURATION

$ env | grep ANSIBLE
ANSIBLE_HOST_KEY_CHECKING=false
##### OS / ENVIRONMENT

Mac OS X El Capitan
##### SUMMARY

The documentation states that listener is an optional item, when it seems it is not.
##### STEPS TO REPRODUCE

Using the following task:

```
- name: Create ELB for sender servers
  local_action:
    module: ec2_elb_lb
    name: "{{ elb_sender }}"
    state: present
    zones: "{{ availability_zones }}"
    tags:
      Name: "{{ elb_sender }}"
      type: "{{ type_sender }}"
      env: "{{ env }}"
    cross_az_load_balancing: "yes"
    security_group_names: "{{ security_group_sender }}"
    wait: yes
```
##### EXPECTED RESULTS

ELB created without listeners as per documentation
##### ACTUAL RESULTS

```
fatal: [localhost -> localhost]: FAILED! => {"changed": false, "failed": true, "msg": "At least one port is required for ELB creation"}
```

**Comment by ansibot** [comment]
@jsdalton, @rickmendes, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by ghost** [comment]
@minac This looks like a boto3 change. Listeners were not required according to the boto2 docs, but are per the boto3 docs. I will get a PR up this weekend that protects against this and updates the documentation.

**Comment by minac** [comment]
Thanks Rick!

## 

Miguel
https://www.migueldavid.eu

**Comment by minac** [comment]
I'm confused. The listeners are mandatory as per AWS, so I don't understand this change. Can you please explain?

**Comment by ryansb** [comment]
This change is just an update to the documentation to clarify that listeners are mandatory. The code is the same, but the error that used to be raised to the user wasn't very helpful.

**Comment by minac** [comment]
Thank you Ryan. Do you know when will this go live on the website?

## 

Miguel
Website http://www.migueldavid.eu - LinkedIn
http://uk.linkedin.com/in/migueldavid/ - Taste Porto
http://www.tasteportofoodtours.com/

**Comment by ryansb** [comment]
It should be live, I think we build those docs daily.

**Comment by minac** [comment]
Hi Ryan,

Listeners still show up as not required in the docs which was the issue I
raised.

## 

Miguel
https://www.migueldavid.eu

**Comment by ghost** [comment]
@minac @ryansb That's my bad. When I made the second change it looks like I reverted the first that had changed the docs. Let me put up another PR to fix that.

**Comment by ghost** [comment]
@minac @ryansb Actually, looking back at my old PR, we cannot make this required in all cases. There are use cases in the module where listeners are optional. So the fix is a code fix that checks in the correct context.

**Comment by minac** [comment]
Interesting, can you share a case when it's not mandatory?

Thanks!

## 

Miguel
https://www.migueldavid.eu

**Comment by ghost** [comment]
@minac It is not mandatory when deleting an ELB.
